docs/guides/00-integration-carto-vl.md

```diff
@@ -0,0 +1,165 @@
+
+    </aside>
+  </main>
+```
+
+
+## Drawing a map with CARTO-VL
+For this guide we are going to use the js code of the [CARTO-VL basic example](https://carto.com/developers/carto-vl/examples/#example-add-carto-dataset-layer at this point you should see a map with an white space on the right.
+
+```html
+<script>
+    const map = new mapboxgl.Map({
+      container: 'map',
+      style: 'https://basemaps.cartocdn.com/gl/voyager-gl-style/style.json',
+      center: [0, 30],
+      zoom: 2,
+      scrollZoom: false,
+      dragRotate: false,
+      touchZoomRotate: false
+    });
+
+    const nav = new mapboxgl.NavigationControl({
+      showCompass: false
+    });
+    map.addControl(nav, 'top-left');
+
+    // Define user
+    carto.setDefaultAuth({
+      user: 'cartovl',
+      apiKey: 'default_public'
+    });
+
+    // Define layer
+    const source = new carto.source.Dataset('ne_10m_populated_places_simple');
+    const viz = new carto.Viz();
+    const layer = new carto.Layer('layer', source, viz);
+
+    layer.addTo(map, 'watername_ocean');
+    layer.on('loaded', hideLoader);
+
+    function hideLoader() {
+      document.getElementById('loader').style.opacity = '0';
+    }
+</script>
+```
+
+## Adding a Category Widget
+
+Adding a Category Widget is as simple as including `<as-category-widget>` tag within our `as-sidebar` element, setting the options you prefer.
+
+ ```html
+<aside class="as-sidebar as-sidebar--right">
+  <as-category-widget
+    id="cities-widget"
+    heading="Populated places"
+    description="Sum of the population grouped by country">
+  </as-category-widget>
+</aside>
+```
+
+### Binding CARTO-VL data
+
+The category widget will not show any data unless we provide them. For this example, we will simply show the list sorted by the maximum population of cities that can be seen on the map.
+
+To do this we will use the [viewportFeatures expression](https://carto.com/developers/carto-vl/reference/#cartoexpressionsviewportfeatures) which allows us to get the list of the visible features in the viewport.
+
+```js
+// We update the viz to get a new @cities variable with the list of visible cities
+const viz = new carto.Viz(`
+  @cities: viewportFeatures($pop_max, $name),
+`);
+```
+
+Then we get a reference to the widget to update the data on every map change:
+
+```js
+const $categoryWidget = document.querySelector('#cities-widget');
+layer.on('updated', updateWidgets);
+```
+
+The updateWidgets function will:
+
+- Extract data from the viz object
+- Give those data a format compatible with the widget `{name, value}`
+- Sort the data to display most populated places first.
+
+
+```js
+function updateWidgets() {
+    $categoryWidget.categories = viz.variables.cities.value
+      .map(feature => ({
+        name: feature.name,
+        value: feature.pop_max,
+      }))
+      .sort((a, b) => b.value - a.value);
+  };
+```
+
+### Reacting to data changes
+
+> Waiting for a new API in CARTO-VL
```

Comment:
Is this related to filtering or to moving the map?

Reply:
non implemented feature in CARTO-VL, actually @Jesus89 had a similar problem at the same time